test/test_redis_connection.rb

```diff
@@ -223,6 +223,18 @@ def with_env_var(var, uri, skip_provider=false)
     end
 
     describe "with REDIS_PROVIDER set" do
+      it "rejects URLs in REDIS_PROVIDER" do
+        uri = 'redis://sidekiq-redis-provider:6379/0'
+
+        ENV['REDIS_PROVIDER'] = uri
+
+        assert_raises RuntimeError do
+          Sidekiq::RedisConnection.__send__(:determine_redis_provider)
```

Comment:
Why don't you want to use the same cool technique with `private unless $TESTING`?